libres/tests/analysis/test_assert_update_viable.cpp

```diff
@@ -0,0 +1,143 @@
+#include <filesystem>
+
+#include <ert/analysis/update.hpp>
+#include <ert/enkf/analysis_config.hpp>
+#include <ert/enkf/state_map.hpp>
+#include <ert/enkf/local_updatestep.hpp>
+#include <ert/enkf/local_ministep.hpp>
+#include <ert/analysis/analysis_module.hpp>
+
+#include <catch2/catch.hpp>
+
+namespace analysis {
+bool assert_update_viable(const analysis_config_type *analysis_config,
+                          const state_map_type *source_state_map,
+                          const int ensemble_size,
+                          const local_updatestep_type *updatestep);
+}
+
+TEST_CASE("asserting_update_viable", "[analysis][prototype]") {
```

Comment:
Nitpick: the function is called `assert_update_viable`
```suggestion
TEST_CASE("assert_update_viable", "[analysis][prototype]") {
```

Reply:
I would go a bit further and say the fuction name should change, there is no assertion being done. it's more of a `validate_smoother_update`

Reply:
What about `is_valid`? 
I think it makes OK sense since it's just used in `update.cpp`.

Reply:
winner winner chicken dinner